index.bs

```diff
@@ -4900,6 +5008,83 @@ as candidates to be employed in a [=registration ceremony=].
+
+At this time, one [=credential property=], which is an [=all-sources credential property=], is defined. It is known abstractly as the <dfn>resident key credential property</dfn>. It is a boolean value indicating whether or not the {{PublicKeyCredential}} returned as a result of a [=registration ceremony=] is a [=resident credential=] or a [=non-resident credential=].  
+
+: Extension identifier
+:: `credProps`
+
+: Operation applicability
+:: [=registration extension|Registration=]
+
+: Client extension input
+:: The Boolean value [TRUE] to indicate that this extension is requested by the [=[RP]=].
+    <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientInputs {
+        boolean credProps;
+    };
+    </xmp>
+
+: Client extension processing
+::  1. Let |credPropsAuthenticatorInput| be the [=authenticator extension input=], created using the {{AuthenticationExtensionsClientInputs/credProps}} [=client extension input=].
+
+    1. When <a href='#CreateCred-InvokeAuthnrMakeCred'>invoking the authenticatorMakeCredential operation</a> include |credPropsAuthenticatorInput| in the |authenticatorExtensions| parameter.
+
+    1. If no [=authenticator=] indicates success, terminate this extension processing and do not return any outputs.
+
+    1. Assert: an [=authenticator=] [=CreateCred-AuthnrSuccess/indicates success=]. 
+
+    1. If <code>[=credentialCreationData/attestationObjectResult=].authData.[=authdataextensions|extensions=][`credProps`][`rk`]</code>:
+
+	<dl class="switch">
+            :  [=map-exists|Exists=]
+            ::  This is the [=authenticator=]'s result. Terminate these extension steps, and do not return any [=client extension output=].
+
+                Note: the [=authenticator=]'s result will be conveyed to the [=[RP]=] within the <code>[=credentialCreationData/attestationObjectResult=]</code>.
+
+            :  [=map-exists|Does not exist=]
+            :: See "Client extension output", below.
+        </dl>
+
+: Client extension output
+:: [=map-set|Set=] <code>[=credentialCreationData/clientExtensionResults=]["{{AuthenticationExtensionsClientOutputs/credProps}}"]</code> to the value of the |requireResidentKey| parameter that was used in the <a href='#CreateCred-InvokeAuthnrMakeCred'>invocation</a> of the [=authenticatorMakeCredential=] operation.
+
+    <xmp class="idl">
+    dictionary CredentialPropertiesOutput {
+        boolean rk;
+    };
+
+    partial dictionary AuthenticationExtensionsClientOutputs {
+        CredentialPropertiesOutput credProps;
+    };
+    </xmp>
+
+: Authenticator extension input
+:: The Boolean value [TRUE], encoded in CBOR (major type 7, value 21).
+
+: Authenticator extension processing
+:: None, other than to report on credential properties in the output.
+
+: Authenticator extension output
+:: If the [=authenticator=] supports this extension it returns a CBOR map with one entry:
+	<pre>
+	CDDL:
+	{
+	  rk: bool
```

Comment:
"rk" key is a string right?

Reply:
The format seems consistent with other CDDL definitions in the spec.

Reply:
Removing authenticator extension component of the credProps extension per WG call on May 1, 2019 so this is now moot.